### PR TITLE
Fix overprovisionning issue

### PR DESCRIPTION
**Pull request by clarsonneur**
Fixing over provisionning concern as reported in #112

### CODE REVIEW

**Comment by KostyaSha** [comment]
Maybe you are right, i were checking this part of code also.

**Comment by KostyaSha** [comment]
test fails :(

File: yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java [hunk 1]
```diff
@@ -66,7 +66,8 @@ public void setEnabled(boolean enabled) {
 
                 int availableCapacity = snapshot.getAvailableExecutors() +
                         snapshot.getConnectingExecutors() +
-                        strategyState.getAdditionalPlannedCapacity();
+                        strategyState.getAdditionalPlannedCapacity() +
+                        strategyState.getPlannedCapacitySnapshot();
```
Comment (KostyaSha) [inline yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java]: I used this https://github.com/jenkinsci/mansion-cloud-plugin/blob/master/src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionNodeProvisionerStrategy.java#L123-L124
@stephenc do you have bug in mansion?

Comment (stephenc) [inline yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java]: Who knows. Mansion is not in my "great" list, only in my "not total crap" list

**Comment by clarsonneur** [comment]
@KostyaSha when you tested, what fails in which context?

**Comment by KostyaSha** [comment]
@clarsonneur see travis test, i restarted it... Or maybe it caused by delay in pull... will restart one more time. As this provisioning and plugin are not ideal i think we can merge, release and try.

**Comment by clarsonneur** [comment]
ok. I saw travis error as well. Looks like it was about a network issue. We got this:
java.net.SocketTimeoutException: Read timed out. I'm going to check the test, to see how it is related.

File: yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java (hunk 1 above)
Comment (KostyaSha) [inline yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java]: @stephenc so no ideas?

**Comment by KostyaSha** [comment]
@clarsonneur test passed.

**Comment by clarsonneur** [comment]
The thing I noticed using the debugger was how counters were updated.

When the provisionning starts, it calls `Apply`, then :

1. Start provisionning ([line 78](https://github.com/clarsonneur/yet-another-docker-plugin/blob/5d684387741e052544ca705c5488ecaf7323b8b3/yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java#L78))
2. Update counters ([line 81](https://github.com/clarsonneur/yet-another-docker-plugin/blob/5d684387741e052544ca705c5488ecaf7323b8b3/yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java#L81) & [line 83](https://github.com/clarsonneur/yet-another-docker-plugin/blob/5d684387741e052544ca705c5488ecaf7323b8b3/yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerProvisioningStrategy.java#L83))
  It updates AdditionalPlannedCapacity counter.
3. Return.

Then jenkins recall `Apply`, and `PlannedCapacitySnapshot` is getting increased with the number found in `AdditionalPlannedCapacity` counter. 
That's why I tried that.

For sure, yeah, mocking that is not an easy part.

Looks like build is success now... :smile:

**Pull request merged**